sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_first_seen_v1/init.sql

```diff
@@ -1,144 +1,33 @@
-  -- I've verified that `date(first_service_timestamp), count(distinct user_id) where did_register = true group by 1`  matches the counts of registrations per day in amplitude.
-first_services_g AS (
-  SELECT
-    * EXCEPT (_n)
-  FROM
-    first_services
-  WHERE
-    _n = 1
-),
-  -- sadly, `entrypoint` is null on registration complete and login complete events.
-  -- this means we have to use first_service_flow to join back on the original source table's flow_id,
-  -- and take the first occurrence of `entrypoint` within the flow that the user first appeared in the service on.
-flows AS (
-  SELECT DISTINCT
-    s.first_service_flow,
-    FIRST_VALUE(f.entrypoint) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_entrypoint,
-    FIRST_VALUE(f.timestamp) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_timestamp,
-    FIRST_VALUE(f.country) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_country,
-    FIRST_VALUE(f.os_name) OVER (
-      PARTITION BY
-        f.flow_id
-      ORDER BY
-        f.`timestamp`
-      ROWS BETWEEN
-        UNBOUNDED PRECEDING
-        AND UNBOUNDED FOLLOWING
-    ) AS first_service_os
+    IF(service IS NULL AND event_type = 'fxa_activity - cert_signed', 'sync', service) AS service,
+    country,
+    os_name,
   FROM
-    first_services_g s
-  INNER JOIN
-    firefox_accounts.fxa_content_auth_oauth_events AS f
-  ON
-    s.first_service_flow = f.flow_id
-  WHERE
-    f.entrypoint IS NOT NULL
-    AND s.first_service_flow IS NOT NULL
-    AND DATE(f.`timestamp`) >= '2019-03-01'
+    `moz-fx-data-shared-prod.firefox_accounts.fxa_content_auth_oauth_events`
+  WHERE DATE(timestamp) = DATE("2022-01-01")
 )
-  -- finally take the entrypoint data and join it back on the other information (os, country etc).
-  -- also, add a row number that indicates the order in which the user signed up for their services.
+
 SELECT
-  s.user_id,
-  s.service,
-  s.first_service_flow,
-  s.did_register,
-  f.first_service_entrypoint AS entrypoint,
-  f.first_service_timestamp,
-  f.first_service_country,
-  f.first_service_os,
-  ROW_NUMBER() OVER (PARTITION BY s.user_id ORDER BY first_service_timestamp_last) AS service_number
-FROM
-  first_services_g s
-LEFT JOIN
-  flows f
-USING
-  (first_service_flow)
+  DATE(MIN(timestamp)) AS first_seen_date,
+  user_id,
+  service,
+  ANY_VALUE(country) AS country,
+  ANY_VALUE(os_name) AS os_name,
+FROM fxa_events
 WHERE
-  first_service_flow IS NOT NULL
+  user_id IS NOT NULL
+  AND service IS NOT NULL
```

Comment:
honestly, I'm not sure. This piece of logic I copied over from the original model. It might not be needed, I will double check tomorrow.

Reply:
with the current fxa_events CTE seems to imply that services could contain null values. I'd say we should keep this filter out because if no service is listed in the row then we can really use it for this model. Same with user_id, user_id implies it's a record for a registered user.

Reply:
just confirmed, we have `61788819529` records with user_id and service fields being null in the fxa_events CTE since `2021-01-01`